retriever/lib/store.js

```diff
@@ -94,41 +94,45 @@ export async function getOwnerByRootCid(env, rootCid) {
    WHERE ir.root_cid = ?
  `
 
-  const result = await (await env.DB.prepare(query).bind(rootCid).all()).results
+  const { results } = await await env.DB.prepare(query).bind(rootCid).all()
 
-  if (!result || result.length === 0) {
+  if (!results || results.length === 0) {
     return {
       error: `Root_cid '${rootCid}' does not exist or may not have been indexed yet.`,
     }
   }
 
-  const withOwner = result.filter((row) => row && row.owner != null)
+  const withOwner = results.filter((row) => row && row.owner != null)
   if (withOwner.length === 0) {
     return {
       error: `Root_cid '${rootCid}' exists but has no associated owner.`,
     }
   }
   const approved = withOwner.find(
     (row) =>
-      typeof row.owner === 'string' &&
-      approvedOwners.includes(String(row.owner)),
+      typeof row.owner === 'string' && approvedOwners.includes(row.owner),
   )
   if (!approved || approved.length === 0) {
     return {
-      error: `Root_cid '${rootCid}' exists but has no associated owner from the approved list.`,
+      error: `Root_cid '${rootCid}' exists but has no approved owner`,
     }
   }
+
+  if (!approved || typeof approved.owner !== 'string') {
+    // Should never be called
+    return { error: `Record has unexpected format ${JSON.stringify(approved)}` }
+  }
```

Comment:
Is this required for TS? Can we remove that dead block?

Reply:
I think it may have been needed while `approved` had type `Record<string, unknown>`. That that it's typed, I think this shouldn't be needed anymore.

Reply:
(I am also surprised `tsc` did not flag this.)